api/src/main/java/com/vimeo/networking2/config/RetrofitSetupModule.kt

```diff
@@ -76,36 +48,54 @@ object RetrofitSetupModule {
      * Creates the object graph for the setup dependencies. After the graph is created, the method
      * return an instance of [Retrofit] which is root of the graph.
      */
-    fun retrofit(serverConfig: ServerConfig, accessToken: String? = null): Retrofit {
+    fun retrofit(serverConfig: ServerConfig, userAgent: String? = null, accessToken: String? = null): Retrofit {
         val interceptors = mutableListOf<Interceptor>()
         if (accessToken?.isNotBlank() == true) {
             interceptors.add(accessTokenInterceptor(accessToken))
         }
-        interceptors.add(userAgentInterceptor)
+        interceptors.add(userAgentInterceptor(userAgent))
         interceptors.add(acceptHeaderInterceptor)
 
         if (serverConfig.customInterceptors?.isNotEmpty() == true) {
             interceptors.addAll(serverConfig.customInterceptors)
         }
+
         val okHttpClient = okHttpClient(serverConfig, interceptors)
         return createRetrofit(serverConfig, okHttpClient)
     }
 
     /**
-     * Create [Retrofit] with OkHttpClient and Moshi.
+     * Interceptor for adding an access token.
      */
-    private fun createRetrofit(serverConfig: ServerConfig, okHttpClient: OkHttpClient): Retrofit =
-        Retrofit.Builder()
-            .baseUrl(serverConfig.baseUrl)
-            .client(okHttpClient)
-            .addConverterFactory(MoshiConverterFactory.create(moshi))
-            .addCallAdapterFactory(ErrorHandlingCallAdapterFactory())
-            .build()
+    private fun accessTokenInterceptor(accessToken: String) =
+        Interceptor { chain ->
+            val request = chain.request()
+            val builder = request.newBuilder()
+
+            if (request.header(AUTHORIZATION_HEADER).isNullOrBlank()) {
+                builder.addHeader(AUTHORIZATION_HEADER, "Bearer $accessToken")
+            }
+            chain.proceed(builder.build())
+        }
 
     /**
-     * Creates a cache for storing Retrofit services.
+     * Create interceptor for adding a user agent.
      */
-    fun retrofitCache(retrofit: Retrofit) = RetrofitServicesCache(retrofit)
+    private fun userAgentInterceptor(userAgent: String?) =
+        Interceptor { chain ->
+            val request = chain.request()
+            val builder = request.newBuilder()
+
+            if (request.header(USER_AGENT_HEADER).isNullOrBlank()) {
+
+                val customUserAgent = userAgent?.let {
```

Comment:
The user agent should be additive to the `USER_AGENT_HEADER_VALUE` rather than completely replacing it. See how `RetrofitSetup.createUserAgent` works in the previous version of the library.